Rework the partitioned table introduction (#11138)
crate/crate: docs/general/ddl/partitioned-tables.rst

      |
      +-- partition 2
            |
            +- shard 0
            |
            +- shard 1

A table becomes a partitioned table by defining :ref:`partition columns
<gloss-partition-column>`.  When a record with a new distinct combination of
values for the configured :ref:`partition columns <gloss-partition-column>` is
inserted, a new partition is created and the document will be inserted into
this partition.

A partitioned table can be queried like a regular table. 

Partitioned tables have the following advantages:

- The number of shards can be changed on the partitioned table, which will then
  change how many shards will be used for the next partition creation. This
  enables one to start out with few shards per partition initially, and scale
  up the number of shards for later partitions once traffic and ingest rates
  increase with the lifetime of an application.

- Partitions can be backed up and restored individually.

- Queries which contain filters in the ``WHERE`` clause which identify a single
  partition or a subset of partitions is less expensive than querying all
  partitions because the shards of the excluded partitions won't have to be
  accessed.

> - Deleting data from a partitioned table is cheap if the optimizer identifies
>   that a full partition can be dropped.

Reviewer: ```suggestion
- Deleting data from a partitioned table is cheap if full partitions are dropped.
```

Maybe meaning the same but slightly easier to read like this? Again fine to leave it as is!

Author: @jayeff I pushed a fixup that takes your suggestion and elaborates a bit further. Please let me know what you think.

Reviewer: Looks good 👍

Partitioned tables have the following disadvantages:

> - If the partitioned columns are badly chosen you can end up with too many

Reviewer: ```suggestion
- If the partition columns are badly chosen you can end up with too many
```

Minor consistency change to reflect the usage at the beginning of the Introduction.

`partitioned columns` is fine too but according to Glossary
> **Partitioned columns** Better known as a partition column.
> https://crate.io/docs/crate/reference/en/master/appendices/glossary.html#gloss-partitioned-column

  shards in the cluster. Affecting the overall stability and performance
  negatively.

- You may end up with empty, stale partitions if delete operations couldn't be
  optimized to drop full partitions. You may have to watch out for this and
  invoke ``DELETE`` statements to target single partitions to clean them up.

> - Some optimizations don't apply to partitioned tables. An example for this is

Author: Not sure if it's worth to include this. I think there's currently only this single optimization that's mentioned as example which doesn't apply.

Author: > What we could do, is add reasons why this is the case

I think it would blow up the introduction to go into the details here, but I think a dedicated section about optimizations could work. 

>  but I think we need some examples / tutorials for some good and bad paritioning examples.

:+1: 
Who's going to write that? :)

Reviewer 2: 🤷🏻‍♀️ hopefully will be a collaboration, but we/I would like to start it :)
